hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/cmd/UpdateCollectionIntervalsCommand.java

```diff
@@ -0,0 +1,200 @@
+
+    public UpdateCollectionIntervalsCommand() {
+        super("Update Collection Intervals", "DMR Nodes");
+    }
+
+    /** @see org.hawkular.agent.monitor.cmd.AbstractResourcePathCommand#createResponse() */
+    @Override
+    protected UpdateCollectionIntervalsResponse createResponse() {
+        return new UpdateCollectionIntervalsResponse();
+    }
+
+    @Override
+    protected BinaryData execute(
+            ModelControllerClient mcc,
+            EndpointService<DMRNodeLocation, DMRSession> endpointService,
+            String modelNodePath,
+            BasicMessageWithExtraData<UpdateCollectionIntervalsRequest> envelope,
+            UpdateCollectionIntervalsResponse response,
+            CommandContext context,
+            DMRSession dmrContext) throws Exception {
+
+        UpdateCollectionIntervalsRequest request = envelope.getBasicMessage();
+        Map<PathAddress, String> metricTypes = filterUpdates(mcc, request.getMetricTypes(), false);
+        Map<PathAddress, String> availTypes = filterUpdates(mcc, request.getAvailTypes(), true);
+
+        if (isEmpty(metricTypes) && isEmpty(availTypes)) {
+            log.debug("Skipping collection interval update, no valid type updates provided.");
+            return null;
+        }
+
+        CompositeOperationBuilder<?> composite = OperationBuilder.composite();
+        addUpdates(composite, metricTypes);
+        addUpdates(composite, availTypes);
+        // Don't let the restart hooks kick in, instead do a deferred restart
+        //composite.allowResourceServiceRestart();
+        composite.byNameOperation("start")
+                .address(PathAddress.parseCLIStyleAddress("/subsystem=hawkular-wildfly-agent/"))
+                .attribute("restart", "true")
+                .attribute("delay", "500") //  enough time to get the response sent back
+                .parentBuilder();
+
+        try {
+            OperationResult<?> opResult = composite.execute(mcc).assertSuccess();
+            setServerRefreshIndicator(opResult, response);
+
+        } catch (DmrApiException e) {
+            log.errorf("Failed to update collection intervals: %s", e.getMessage());
+            throw e;
+        }
+
+        return null;
+    }
+
+    private Map<PathAddress, String> filterUpdates(ModelControllerClient mcc, Map<String, String> updates,
+            boolean isAvail) {
+
+        if (isEmpty(updates)) {
+            return null;
+        }
+
+        Map<PathAddress, String> result = new HashMap<>();
+
+        for (Map.Entry<String, String> entry : updates.entrySet()) {
+            String typeId = entry.getKey();
+            String interval = entry.getValue();
+
+            PathAddress metricType = null;
+            try {
+                String path = isAvail ? toAvailPath(typeId) : toMetricPath(typeId);
+                metricType = PathAddress.parseCLIStyleAddress(path);
```

Comment:
This won't work on EAP6, please to use this: WildflyCompatibilityUtils.parseCLIStyleAddress

`WildflyCompatibilityUtils.parseCLIStyleAddress(path)`